apm-sniffer/apm-sdk-plugin/solrj-7.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/solrj/SolrClientInterceptor.java

```diff
@@ -0,0 +1,198 @@
+            if (update.getAction() != null) {
+                action = update.getAction().name();
+
+                if (update.getAction() == AbstractUpdateRequest.ACTION.COMMIT) {
+                    span = getSpan(collection, request.getPath(), action, instance.getRemotePeer());
+
+                    span.tag(SolrjTags.TAG_COMMIT, params.get(UpdateParams.COMMIT, "true"));
+                    span.tag(SolrjTags.TAG_SOFT_COMMIT, params.get(UpdateParams.SOFT_COMMIT, ""));
+                } else if (update.getAction() == AbstractUpdateRequest.ACTION.OPTIMIZE) {
+                    span = getSpan(collection, request.getPath(), action, instance.getRemotePeer());
+
+                    span.tag(SolrjTags.TAG_OPTIMIZE, params.get(UpdateParams.OPTIMIZE, "true"));
+                    span.tag(SolrjTags.TAG_MAX_OPTIMIZE_SEGMENTS, params.get(UpdateParams.MAX_OPTIMIZE_SEGMENTS, "1"));
+                }
+            } else {
+                if (update instanceof UpdateRequest) {
+                    UpdateRequest ur = (UpdateRequest) update;
+                    List<SolrInputDocument> documents = ur.getDocuments();
+                    if (documents == null) {
+                        action = "DELETE";
+
+                        span = getSpan(collection, request.getPath(), action, instance.getRemotePeer());
+                        List<String> deleteBy = ur.getDeleteById();
+                        if (deleteBy != null && !deleteBy.isEmpty()) {
+                            span.tag(SolrjTags.TAG_DELETE_TYPE, "deleteByIds");
+                            span.tag(SolrjTags.TAG_DELETE_VALUE, deleteBy.toString());
+                        }
+                        List<String> deleteQuery = ur.getDeleteQuery();
+                        if (deleteQuery != null && !deleteQuery.isEmpty()) {
+                            span.tag(SolrjTags.TAG_DELETE_TYPE, "deleteByQuery");
+                            span.tag(SolrjTags.TAG_DELETE_VALUE, deleteQuery.toString());
+                        }
+                    } else {
+                        span = getSpan(collection, request.getPath(), action, instance.getRemotePeer());
+                        span.tag(SolrjTags.TAG_DOCS_SIZE, String.valueOf(documents.size()));
+                        span.tag(SolrjTags.TAG_COMMIT_WITHIN, String.valueOf(update.getCommitWithin()));
+                    }
+                }
+            }
+            span.tag(SolrjTags.TAG_QT, params.get(CommonParams.QT, "/update"));
+            span.tag(SolrjTags.TAG_ACTION, action);
+        } else if (request instanceof QueryRequest) {
+            String operatorName = String.format("solrJ/%s%s", collection, request.getPath());
+            span = ContextManager.createExitSpan(operatorName, instance.getRemotePeer())
+                    .setComponent(ComponentsDefine.SOLRJ)
+                    .setLayer(SpanLayer.DB);
+            span.tag(SolrjTags.TAG_QT, params.get(CommonParams.QT, "/select"));
+        }
+
```

Comment:
Above codes include a lot of `instanceof`, which would be an impact for performance, especially in JRE 6-8. If possible, please consider a better way to do this.